src/components/NewAnalysisModal.js

```diff
@@ -1,37 +1,106 @@
 
 export const NewAnalysisModal = Utils.withDisplayName('NewAnalysisModal')(
-  ({ isOpen, onDismiss, onSuccess, runtimes, apps, galaxyDataDisks, refreshRuntimes, refreshApps, workspace, persistentDisks, workspace: { workspace: { namespace, bucketName, name: workspaceName } } }) => {
+  ({ isOpen, onDismiss, onSuccess, uploadFiles, openUploader, runtimes, apps, galaxyDataDisks, refreshRuntimes, refreshApps, refreshAnalyses, analyses, workspace, persistentDisks, workspace: { workspace: { namespace, bucketName, name: workspaceName } } }) => {
     const [viewMode, setViewMode] = useState(undefined)
     const [busy, setBusy] = useState()
+    const [notebookKernel, setNotebookKernel] = useState('python3')
+    const [nameTouched, setNameTouched] = useState(false)
+    const [analysisName, setAnalysisName] = useState('')
+    const [currentTool, setCurrentTool] = useState(undefined)
 
-    const renderNewRuntimeModal = tool => h(NewRuntimeModalBase, {
-      isOpen: viewMode === NEW_JUPYTER_MODE || viewMode === NEW_RSTUDIO_MODE,
+
+    const currentRuntime = getCurrentRuntime(runtimes)
+    const currentRuntimeTool = currentRuntime?.labels?.tool
+    const currentApp = getCurrentApp(apps)
+
+    const resetView = () => {
+      setViewMode(undefined)
+      setAnalysisName('')
+      setCurrentTool(undefined)
+      setNotebookKernel('python3')
+    }
+
+    const enterNextViewMode = (currentTool, baseViewMode = viewMode) => {
+      const doesCloudEnvForToolExist = currentRuntimeTool === currentTool || (currentApp && currentTool === tools.galaxy.label)
+
+      Utils.switchCase(baseViewMode,
+        [NEW_ANALYSIS_MODE, () => Utils.cond(
+          [doesCloudEnvForToolExist, () => {
+            resetView()
+            onSuccess()
+          }],
+          [!doesCloudEnvForToolExist && currentRuntime && isRuntimeDeletable(currentRuntime), () => setViewMode(NEW_ENVIRONMENT_MODE)],
+          [!doesCloudEnvForToolExist && !currentRuntime, () => setViewMode(NEW_ENVIRONMENT_MODE)],
+          [!doesCloudEnvForToolExist && currentRuntime && !isRuntimeDeletable(currentRuntime), () => {
+            resetView()
+            onSuccess()
+          }]
+        )],
+        [NEW_ENVIRONMENT_MODE, () => {
+          resetView()
+          onSuccess()
+        }],
+        [Utils.DEFAULT, () => Utils.cond(
+          [currentTool === tools.RStudio.label || currentTool === tools.Jupyter.label, () => setViewMode(NEW_ANALYSIS_MODE)],
+          [currentTool === tools.galaxy.label && !currentApp, () => setViewMode(NEW_ENVIRONMENT_MODE)],
+          [currentTool === tools.galaxy.label && currentApp, () => resetView()] //This shouldn't be possible, as the button is disabled in this case
+        )]
+      )
+    }
+
+    const setPreviousViewMode = () => Utils.switchCase(viewMode,
+      [NEW_ANALYSIS_MODE, () => resetView()],
+      [NEW_ENVIRONMENT_MODE, () => resetView()],
+      [Utils.DEFAULT, () => undefined]
+    )
```

Comment:
As far as I can tell, viewMode is one of those values or undefined, in which case this function can be simplified to:
```suggestion
    const setPreviousViewMode = () => !!viewMode && resetView()
```
If there is only one usage, it should be inlined for clarity.